trace/tests/system/flask/flask_system_test.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2017, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import random
+import requests
+import shlex
+import signal
+import subprocess
+import time
+import uuid
+
+import unittest
+
+PROJECT = os.environ.get('GCLOUD_PROJECT_PYTHON')
+
+
+def wait_app_to_start():
+    """Wait the application to start running."""
+    cmd = 'until nc -z -v -w30 127.0.0.1 8080; do sleep 2; done'
```

Comment:
Is there a way to make this fail after a certain timeout?  It looks like it would loop forever?

I had used `wget --retry-connrefused --tries=5 http://localhost:$PORT/` elsewhere, which returns with an error code after 5 tries (~60 seconds I think).

Reply:
Thanks for the suggestion! Changed.